Refactor: category section and sidebar filtering

Zyruks opened the pull request:
## Changes Made 🎉

- **feat:** Added new `CategorySection` component to replace `RaritySection` for generic filtering by `Rarity` or `Alphabet`.
- **feat:** Integrated `FilterMode` and `SortOrder` into `UserSidebar` to allow users to filter and sort their collection.
- **feat:** Implemented the `CardStack` component for displaying cards with stacking effects for duplicates.
- **fix:** Corrected sorting logic in `getCategoryCriteria` to properly handle ascending/descending order for both Rarity and Alphabet filters.
- **refactor:** Updated `UserCollectionPage` to utilize the new category-based filtering and sorting.
- **style:** Updated `global.css` to adjust the default color scheme for light and dark modes.

### Describe Changes

This PR refactors the collection filtering and sorting system to use a more flexible `CategorySection` component. Users can now filter cards by rarity or alphabet, and sort them in ascending or descending order via the `UserSidebar`. The `CardStack` component was introduced to improve card display, especially when multiple duplicates exist. The sorting logic in `getCategoryCriteria` was also fixed to address incorrect order for rarity.



## Checklist ✅

- [x] My code follows the code style of this project.
- [x] I have updated the documentation where necessary.
- [x] The changes add new functionality and fix an existing issue with sorting logic.

Zyruks committed:
feat(dependencies): ✨ add new libraries for UI components and icons

- Added `@radix-ui/react-select` for select UI component.
- Added `lucide-react` for icon support.

Zyruks committed:
feat(components): ✨ add Select component using Radix UI and lucide-react icons

- Implemented `Select`, `SelectTrigger`, `SelectItem`, and other subcomponents.
- Integrated Radix UI's select elements for enhanced accessibility.
- Used `lucide-react` icons for visual improvements.

Zyruks committed:
feat(utils): ✨ add getCategoryCriteria utility for filtering and sorting categories

- Implemented `getCategoryCriteria` function to handle filtering by `Rarity` or `Alphabetical` mode.
- Supports sorting categories in both ascending and descending order based on `SortOrder`.

Zyruks committed:
feat(constants): ✨ add FilterMode and SortOrder enums, and ALPHABET constant

- Added `FilterMode` enum with `Rarity` and `Alphabet` options to `card.ts`.
- Added `SortOrder` enum for `Ascending` and `Descending` sorting to `card.ts`.
- Added `ALPHABET` constant for alphabetical filtering in `general.ts`.

Zyruks committed:
style(global): 💄 update light mode colors to dark theme defaults

- Updated root-level CSS variables to use dark mode values as default.
- Removed redundant dark mode color definitions for background, foreground, and other elements.

Zyruks committed:
refactor(components): ♻️ rename and update RaritySection to CategorySection for generic filtering

- Renamed `RaritySection` to `CategorySection` for broader category-based filtering.
- Updated logic to support filtering by `FilterMode` (Alphabetical or Rarity) and sorting by `SortOrder`.
- Utilized `getCategoryCriteria` to dynamically generate category criteria.

Zyruks committed:
feat(sidebar): ✨ add filter and sort functionality to user sidebar

- Integrated `Select` components for filtering cards by `FilterMode` (Rarity or Alphabet).
- Added sorting functionality for cards with `SortOrder` (Ascending or Descending).
- Refactored the sidebar to dynamically calculate owned card counts by rarity and foil status.

Zyruks committed:
feat(components): ✨ add CardStack component for displaying card stacks with duplicates

- Implemented `CardStack` component to display individual cards with stacking effects for duplicates.
- Supports displaying both foil and non-foil versions of cards based on user ownership.
- Added hover animations for stacked card visuals.

Zyruks committed:
refactor(collection): ♻️ update UserCollectionPage with category-based filtering and sorting

- Replaced `RaritySection` with `CategorySection` to support filtering by `FilterMode` (Rarity or Alphabet).
- Added sorting functionality with `SortOrder` (Ascending or Descending).
- Updated `UserSidebar` to manage filter and sort options.
- Refactored data handling for user collections and card filtering.

Zyruks committed:
refactor: 💄 update style, spacing and colors

Zyruks committed:
fix(utils): 🐛 correct sorting logic in getCategoryCriteria for Rarity filter mode

- Fixed issue where Rarity categories were incorrectly ordered when using ascending sort.
- Updated `getCategoryCriteria` to handle Rarity and Alphabet sorting with distinct logic.

Zyruks commented on apps/nextjs/package.json:
```diff
@@ -31,6 +32,7 @@
     "clsx": "^2.1.1",
     "eslint": "^9.11.1",
     "eslint-config-next": "14.2.15",
+    "lucide-react": "^0.453.0",
```
El select usa lucide, luego podemos ver como agregar los nuestro de momento son dos chevron

Zyruks committed:
refactor: 🚚 update activeFIlter to activeCategory

serudda commented on apps/nextjs/src/common/constants/general.ts:
```diff
@@ -1,3 +1,5 @@
+export const ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
```
esto para?

serudda reviewed (commented):
Muchisimo mas limpio 🫰

Zyruks committed:
feat(category-criteria): ✨ enhance category filtering with card names

- Refactor `getCategoryCriteria` to accept an array of cards for dynamic filtering
- Introduce unique first-letter extraction from card names for non-Rarity filter mode
- Adjust sorting logic to handle cards instead of static alphabet

Zyruks committed:
refactor(category-section): ♻️ pass allCards to getCategoryCriteria for dynamic filtering

- Update `useMemo` hook to include `allCards` as a dependency
- Ensure `getCategoryCriteria` receives card data for proper filtering

Zyruks committed:
chore: 🔥 remove no necessary constant
